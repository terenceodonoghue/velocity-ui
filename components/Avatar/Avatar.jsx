/** @jsx jsx */
import { jsx } from '@emotion/core';
import * as S from './Avatar.styles';

const Round = props => <S.Round {...props} />;
const Square = props => <S.Square {...props} />;

export default {
  Round,
  Square,
};
