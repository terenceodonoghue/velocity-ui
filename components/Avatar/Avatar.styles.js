/* eslint-disable import/prefer-default-export */
import styled from '@emotion/styled';

export const Round = styled.img`
  border-radius: 50%;
`;
