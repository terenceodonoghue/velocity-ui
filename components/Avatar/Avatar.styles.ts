import { css, InterpolationWithTheme } from '@emotion/core';
import { Sizable, Theme } from '~/types';

export const round = ({
  size,
}: Sizable): InterpolationWithTheme<Theme> => () => css`
  border-radius: 50%;
  height: ${size}px;
  width: ${size}px;
`;

export const square = ({
  size,
}: Sizable): InterpolationWithTheme<Theme> => () => css`
  border-radius: 6px;
  height: ${size}px;
  width: ${size}px;
`;
