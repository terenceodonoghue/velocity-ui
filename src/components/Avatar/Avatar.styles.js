import styled from '@emotion/styled';

export const Round = styled.img`
  border-radius: 50%;
`;

export const Square = styled.img`
  border-radius: 6px;
`;
