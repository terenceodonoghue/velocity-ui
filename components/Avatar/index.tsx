/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FunctionComponent } from 'react';
import { AvatarProps } from '~/types';
import * as css from './Avatar.styles';

const Avatar: FunctionComponent<AvatarProps> = ({
  alt,
  round,
  size,
  ...props
}) => (
  <img
    alt={alt}
    css={round ? css.round({ size }) : css.square({ size })}
    {...props}
  />
);

export default Avatar;
