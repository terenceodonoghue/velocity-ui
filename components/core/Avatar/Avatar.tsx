import React, { FunctionComponent } from 'react';
import * as css from './Avatar.styles';
import type { AvatarProps } from './Avatar.types';

const Avatar: FunctionComponent<AvatarProps> = ({
  alt,
  size,
  variant = 'rounded',
  ...props
}) => (
  <img
    alt={alt}
    css={[
      variant === 'rounded' ? css.rounded : css.square,
      { height: size, width: size },
    ]}
    data-testid="avatar"
    {...props}
  />
);

export default Avatar;
