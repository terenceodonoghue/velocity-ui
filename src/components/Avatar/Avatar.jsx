import React from 'react';
import * as S from './Avatar.styles';

const Round = (props) => <S.Round {...props} />;

const Square = (props) => <S.Square {...props} />;

export default {
  Round,
  Square,
};
