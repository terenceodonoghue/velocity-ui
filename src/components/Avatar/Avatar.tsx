/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FunctionComponent, ImgHTMLAttributes } from 'react';
import * as css from './Avatar.styles';

export const Round: FunctionComponent<ImgHTMLAttributes<HTMLImageElement>> = ({
  alt,
  ...props
}) => <img alt={alt} css={css.round} {...props} />;

export const Square: FunctionComponent<ImgHTMLAttributes<HTMLImageElement>> = ({
  alt,
  ...props
}) => <img alt={alt} css={css.square} {...props} />;
