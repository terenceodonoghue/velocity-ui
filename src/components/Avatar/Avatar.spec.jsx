import React from 'react';
import Avatar from './Avatar';

describe('Avatar', () => {
  describe('Round', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(<Avatar.Round src="https://goo.gl/fbAQLP" />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Square', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(<Avatar.Square src="https://goo.gl/fbAQLP" />)
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
