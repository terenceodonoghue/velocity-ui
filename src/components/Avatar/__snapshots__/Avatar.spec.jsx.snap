// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Round matches snapshot 1`] = `
.emotion-0 {
  border-radius: 50%;
}

<img
  className="emotion-0 emotion-1"
  src="https://goo.gl/fbAQLP"
/>
`;

exports[`Avatar Square matches snapshot 1`] = `
.emotion-0 {
  border-radius: 6px;
}

<img
  className="emotion-0 emotion-1"
  src="https://goo.gl/fbAQLP"
/>
`;
